### DISCUSSION TITLE
Ping/pong on websocket every few seconds forever

### DISCUSSION THREAD

**Discussion by sklibanov312** [zauberzeug/nicegui#2075] [Q&A] [answered]
### Description

I'm trying to minimize unnecessary traffic to my web server and I see that the nicegui server and client are constantly sending ping/pong messages along their websocket every few seconds.

To reproduce:

1. Open Chrome's Developer Tools, look in the Network tab
2. Open https://nicegui.io/documentation/label or any other nicegui app page. I chose that one because it's so simple.
3. Look in the Network tab and find the socket.io _nicegui_ws/.... websocket. Note that its messages look like this if you just do nothing to the page:
![image](https://github.com/zauberzeug/nicegui/assets/111643872/8120676d-def5-4d84-a0f1-3985911defab)
over and over, forever. 

Is there a specific reason that the default values for these ping intervals that are this chatty? With the current defaults, a computer that someone simply stops using and walks away from with an idle page will continue to eat your server resources. I'd like to be able to tune these parameters.

Is there a document that describes the tuning of these parameters? Things like:

* How often pings are sent
* How long until a reply is expected
* How high the exponential backoff to re-connect can go in seconds

I understand that a total disconnect would be difficult, although one *could* imagine pickling the specific client's server state to a database after a long-enough idle period...

Any advice on these topics?

**Comment by falkoschindler** [comment, marked as answer]
Hi @sklibanov312, the ping interval is basically controlled by the `reconnect_timeout` in `ui.run` which defaults to 3 seconds. On nicegui.io, however, we set it to 10 seconds. This is the time the server waits for the client to reconnect after the connection got lost, before the client is removed and resources are freed up.

The ping interval is set in nicegui.py like this:
https://github.com/zauberzeug/nicegui/blob/59546fb99286898d3d4ff128d9d5ce03dce9fab4/nicegui/nicegui.py#L90-L91

For a reconnect timeout of 10 seconds the ping interval is 8 seconds, just as you measured empirically.

To reduce the ping interval, the easiest thing would be to increase the reconnect timeout via `ui.run`. But you can also fine-tune the ping interval in `sio.eio` like this:
```py
from nicegui import core
core.sio.eio.ping_interval = ...
```
But it shouldn't exceed the reconnect timeout.
